workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
+
+		CrossRef relation types: https://www.crossref.org/documentation/schema-library/markup-guide-metadata-segments/relationships/
+
+		string id - persistent unique ID for an entity. Should be in the format
+			<database name>:<identifier within database>
+			e.g.	DOI:10.46936/10.25585/60000745
+				GO:0005456
+				HGNC:7470
+		string description (optional) - description of that entity
+		string relationship_type (optional) - the relationship between the ID and some
+			other entity.
+			For example, when a PermanentID class is used to represent objects in the
+			CreditMetadata field 'related_identifiers', the 'relationship_type' field
+			captures the relationship between the CreditMetadata and this ID.
+
+			Valid relationship_type values:
+				"Datacite:Cites"
+				"Datacite:Compiles"
+				"Datacite:Continues"
+				"Datacite:Describes"
+				"Datacite:Documents"
+				"Datacite:HasMetadata"
+				"Datacite:HasPart"
+				"Datacite:HasVersion"
+				"Datacite:IsCitedBy"
+				"Datacite:isCompiledBy"
+				"Datacite:IsContinuedBy"
+				"Datacite:IsDerivedFrom"
+				"Datacite:IsDescribedBy"
+				"Datacite:IsDocumentedBy"
+				"Datacite:IsIdenticalTo"
+				"Datacite:IsMetadataFor"
+				"Datacite:IsNewVersionOf"
+				"Datacite:IsOriginalFormOf"
+				"Datacite:IsPartOf"
+				"Datacite:IsPreviousVersionOf"
+				"Datacite:IsPublishedIn"
+				"Datacite:IsReferencedBy"
+				"Datacite:IsRequiredBy"
+				"Datacite:IsReviewedBy"
+				"Datacite:IsSourceOf"
+				"Datacite:IsSupplementTo"
+				"Datacite:IsSupplementedBy"
+				"Datacite:IsVariantFormOf"
+				"Datacite:IsVersionOf"
+				"Datacite:Obsoletes"
+				"Datacite:References"
+				"Datacite:Requires"
+				"Datacite:Reviews"
+				"CrossRef:BasedOnData"
+				"CrossRef:Finances"
+				"CrossRef:HasComment"
+				"CrossRef:HasDerivation"
+				"CrossRef:HasExpression"
+				"CrossRef:HasFormat"
+				"CrossRef:HasManifestation"
+				"CrossRef:HasManuscript"
+				"CrossRef:HasPreprint"
+				"CrossRef:HasRelatedMaterial"
+				"CrossRef:HasReply"
+				"CrossRef:HasReview"
+				"CrossRef:HasTranslation"
+				"CrossRef:IsBasedOn"
+				"CrossRef:IsBasisFor"
+				"CrossRef:IsCommentOn"
+				"CrossRef:IsDataBasisFor"
+				"CrossRef:IsExpressionOf"
+				"CrossRef:IsFinancedBy"
+				"CrossRef:IsFormatOf"
+				"CrossRef:IsManifestationOf"
+				"CrossRef:IsManuscriptOf"
+				"CrossRef:IsPreprintOf"
+				"CrossRef:IsRelatedMaterial"
+				"CrossRef:IsReplacedBy"
+				"CrossRef:IsReplyTo"
+				"CrossRef:IsReviewOf"
+				"CrossRef:IsSameAs"
+				"CrossRef:IsTranslationOf"
+				"CrossRef:Replaces"
+	*/
+	typedef structure {
+		string id;
+		string description;
+		string relationship_type;
+	} PermanentID;
+
+	/* Title
+		Represents the title or name of a resource.
+
+		The 'title_string' field is required; if no value is supplied for 'title_type', it
+		defaults to 'title'. If the 'title_type' is set to 'translated_title', the appropriate
+		BCP-47 tag must be supplied in the 'title_language' field.
```

Comment:
I don't think this constraint is coded in the class is it?

Reply:
Remedied: https://github.com/kbase/workspace_deluxe/pull/659

Reply:
:+1: